### ISSUE TITLE
Mocha 8: Failing async test exits mocha watch mode on reload

### ISSUE THREAD

**Issue by mantoni** [mochajs/mocha#4344]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend that you _not_ install Mocha globally.

### Description

`mocha --watch` exists on second run if async test fails.

### Steps to Reproduce

With this test:

```js
describe('mocha watch issue', () => {

  it('async', (done) => {
    setTimeout(() => {
      throw new Error('Fail');
    }, 1);
  });

});
```

Running:

```bash
mocha --watch
```

And then touching the test file to trigger a second run of the test.

**Expected behavior:** [What you expect to happen]

The test fails on the second run exactly like on the first run and mocha keeps watching.

**Actual behavior:** [What actually happens]

Mocha exits on the second test run.

**Reproduces how often:** [What percentage of the time does it reproduce?]

100%

### Versions

- The output of `mocha --version` and `node node_modules/.bin/mocha --version`: 8.0.1
- The output of `node --version`: v14.4.0
- Your operating system
  - name and version: macOS Catalina 10.15.5
  - architecture (32 or 64-bit): 64
- Your shell (e.g., bash, zsh, PowerShell, cmd): fish
- Your browser and version (if running browser tests): –
- Any third-party Mocha-related modules (and their versions): –
- Any code transpiler (e.g., TypeScript, CoffeeScript, Babel) being used (and its version): –

### Additional Information

I have reported the very same issue for mocha 7 in #4143 and it got fixed. Maybe it's the same issue that somehow found its way back 😄

**Comment by boneskull** [comment]
Yes, we've been monkeying with that code path, trying to reduce the chance we leak `uncaughtException` listeners.

**Comment by boneskull** [comment]
cc @juergba who may have interest in this one.  @mantoni if it's blocking you, adding a test to reproduce this in a PR would be helpful to get us moving.  `test/integration/options/watch.spec.js` seems reasonable, and add a fixture in `test/integration/fixtures/options/watch/`

**Comment by indieisaconcept** [comment]
Hit this issue myself recently. I did a little digging & issue seems to be caused by the cloning the suite on L40. If I remove this hooks fire as expected.

https://github.com/mochajs/mocha/blob/02bdb6bc6c029fb0c01389b27797d00faa76ddde/lib/cli/watch-run.js#L38-L66

**Comment by indieisaconcept** [comment]
FYI looks like `suite.clone()` doesn't copy over any of the hooks.

https://github.com/mochajs/mocha/blob/7d3151d08a3082dd022116d00234caf1600a71b5/lib/suite.js#L117-L127

I did a quick local patch to add these & watch worked as expected.

**Comment by boneskull** [comment]
I think this is closed by #4382.  Please let me know if it isn't.
